src/lib/libmints/deriv.cc

```diff
@@ -747,14 +751,16 @@ SharedMatrix Deriv::compute()
     corr->add(x_contr_);
     corr->add(tpdm_contr_);
     if (reference_separate) {
-        gradient_->add(x_ref_contr_);
-        gradient_->add(opdm_ref_contr_);
-        gradient_->add(tpdm_ref_contr_);
-        SharedMatrix scf_gradient(gradient_->clone());
-        scf_gradient->set_name("Reference Gradient");
-        scf_gradient->print_atom_vector();
-        wfn_->reference_wavefunction()->set_gradient(scf_gradient);
-        corr->print_atom_vector();
+        if ( x_ref_contr_ && opdm_ref_contr_ && tpdm_ref_contr_) {
```

Comment:
I think it'd still be good to rework this logic on cause rather than effect as previously discussed, https://github.com/psi4/psi4/pull/386#discussion_r64315184